src/main/java/mil/dds/anet/database/PositionDao.java

```diff
@@ -170,134 +167,204 @@ public int updateInternal(Position p) {
+            + " AND \"personUuid\" IS NULL AND \"endedAt\" IS NULL")
+        .bind("positionUuid", positionUuid).map(new PersonPositionHistoryMapper()).findFirst()
+        .orElse(null);
+
+    if (emptyPersonPosition == null) {
+      numRows += insertPersonEmptyInPosition(positionUuid);
+    }
+
+    numRows += deletePersonEmptyInPositionEmpty();
+    if (numRows > 0) {
+      return 1;
+    }
+    return numRows;
+  }
+
+  @InTransaction
+  public int removePositionFromPerson(String personUuid) {
+    int numRows = 0;
+    Instant now = Instant.now();
+
+    PersonPositionHistory currentPositionPerson = getDbHandle()
+        .createQuery("/* personEmptyPosition.find */ SELECT * FROM \"peoplePositions\""
+            + " WHERE \"personUuid\" = :personUuid AND \"positionUuid\" IS NOT NULL AND "
+            + " \"endedAt\" IS NULL")
+        .bind("personUuid", personUuid).map(new PersonPositionHistoryMapper()).findFirst()
+        .orElse(null);
+
+    numRows += getDbHandle()
+        .createUpdate(
+            "/* positionRemovePerson.end */ UPDATE \"peoplePositions\" SET \"endedAt\" = :endedAt "
+                + " WHERE \"personUuid\" = :personUuid AND \"positionUuid\" IS NOT NULL AND "
+                + " \"endedAt\" IS NULL")
+        .bind("personUuid", personUuid).bind("endedAt", DaoUtils.asLocalDateTime(now)).execute();
+
+    if (currentPositionPerson != null) {
+      removePersonFromPosition(currentPositionPerson.getPositionUuid());
+    }
+
+    PersonPositionHistory emptyPositionPerson = getDbHandle()
+        .createQuery("/* personEmptyPosition.find */ SELECT * FROM \"peoplePositions\""
+            + " WHERE \"personUuid\" = :personUuid "
+            + " AND \"positionUuid\" IS NULL AND \"endedAt\" IS NULL")
+        .bind("personUuid", personUuid).map(new PersonPositionHistoryMapper()).findFirst()
+        .orElse(null);
+
+    if (emptyPositionPerson == null) {
+      numRows += insertPositionEmptyInPerson(personUuid);
+    }
+
+    numRows += deletePersonEmptyInPositionEmpty();
+    if (numRows > 0) {
+      return 1;
+    }
+    return numRows;
+  }
+
+  @InTransaction
+  public int updatePersonFromPosition(String positionUuid, String loserUuid, String winnerUuid) {
+    int numRows = 0;
+    Instant now = Instant.now();
+    numRows += getDbHandle()
+        .createUpdate("/* position.update */ UPDATE positions "
+            + "SET \"currentPersonUuid\" = :personUuid, \"updatedAt\" = :updatedAt "
+            + "WHERE uuid = :positionUuid")
+        .bind("personUuid", winnerUuid).bind("updatedAt", DaoUtils.asLocalDateTime(now))
+        .bind("positionUuid", positionUuid).execute();
+
+    numRows += updatePersonPositionHistory(loserUuid, winnerUuid);
+
+    numRows += deletePersonEmptyInPositionEmpty();
+    if (numRows > 0) {
+      return 1;
+    }
+    return numRows;
+  }
+
+  @InTransaction
+  public int updatePersonPositionHistory(String loserUuid, String winnerUuid) {
```

Comment:
The position (history) manipulation methods (`cleanPosition, insertPersonEmptyInPosition, insertPositionEmptyInPerson, deletePersonEmptyInPositionEmpty, updatePersonFromPosition, updatePersonPositionHistory`) should have clearer names. Think on an abstract level which methods you need (one to end the 'current' position history record, one to create a new unended position history record). We also *never* delete a position history record (unless the position itself is deleted), so the method `deletePersonEmptyInPositionEmpty` should be unnecessary.

Reply:
The method names have been replaced with explanatory statements about what the function does.


With **deletePersonEmptyInPositionEmpty(deletePeoplePositionsWithoutPositionAndPerson)** 
The excess purposeless records in the table were cleared.
The aim is to prevent the table from growing constantly with unnecessary records.

Reply:
If the aim is to do a one-time cleanup of the existing database, this should be written as a migration (SQL statement in `src/main/resources/migrations.xml`) and not in business logic.

Reply:
The method in business logic has been removed.
SQL script added to migrations.xml file.